plugins/modules/tanzu_mission_control_secret.py

```diff
@@ -0,0 +1,798 @@
+        registry_password=dict(type="str", no_log=True),
+        secret_type=dict(
+            type="str",
+            choices=["opaque", "docker_config"],
+            default="opaque",
+        ),
+        data=dict(type="dict", default={}),
+        export=dict(type="bool", default=False),
+        state=dict(
+            type="str", default="present", choices=["present", "absent", "update"]
+        ),
+    )
+
+    return AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ["cluster_group", "cluster_name"],
+            ["cluster_group", "management_cluster_name"],
+            ["cluster_group", "provisioner_name"],
+            ["data", "registry_host"],
+            ["data", "registry_username"],
+            ["data", "registry_password"],
+        ],
+        required_one_of=[
+            ["cluster_name", "cluster_group"],
+            ["data", "registry_host"],
+            ["data", "registry_username"],
+            ["data", "registry_password"],
+        ],
+        required_together=[
+            ["cluster_name", "management_cluster_name", "provisioner_name"],
+            ["registry_host", "registry_username", "registry_password"],
+        ],
+    )
+
+
+def create_dockerconfig(registry_host, registry_username, registry_password):
+    payload = {
+        "auths": {
+            registry_host: {
+                "username": registry_username,
+                "password": registry_password,
+                "auth": b64encode(
+                    "{}:{}".format(registry_username, registry_password).encode()
+                ).decode(),
+            }
+        }
+    }
+    return {".dockerconfigjson": json.dumps(payload)}
+
+
+def main():
+    module = load_module()
+
+    def key_or_none(param_name):
+        if param_name in module.params:
+            return module.params[param_name]
+        return None
+
+    state = module.params["state"]
+    api_host = module.params["api_host"]
+    api_token = module.params["api_token"]
+    cluster_name = key_or_none("cluster_name")
+    cluster_namespace = module.params["cluster_namespace"]
+    management_cluster_name = key_or_none("management_cluster_name")
+    provisioner_name = key_or_none("provisioner_name")
+    secret_name = module.params["secret_name"]
+    data = module.params["data"]
+    cluster_group = key_or_none("cluster_group")
+    secret_type = module.params["secret_type"]
+    export = module.params["export"]
+
+    try:
+        access_token = exchange_token(api_token)
+        if access_token is None:
+            return module.fail_json(msg="Failed to exchange token, check credentials")
+    except HTTPError as e:
+        return module.fail_json(msg=e.read().decode())
+
+    tmc_secret = None
+
+    if cluster_name is not None:
+        tmc_secret = TMCClusterSecret(
+            str(management_cluster_name),
+            str(provisioner_name),
```

Comment:
Are you sure you want to these parameters to be the string `"None"` (different from the value `None`), when they are not passed to the module?